### PR TITLE
cam6_4_047: update config component to handle new compset names

### LINKED ISSUE #1077: config_component.xml updates needed
### Issue Type

Infrastructure Update

### Issue Description

New compset long names have C or E designations in the date field to indicate concentration or emissions driven (eg 1850C, HISTC) cam's config_component.xml needs to be updated to recognize the new patterns.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

### LINKED ISSUE #1189: O3 is not an invariant
### Issue Type

Infrastructure Update

### Issue Description

I am trying to create a fully active compset using CAM70%MT%CT4S and I am getting an error from cam 
```
trcdata_init: file%has_ps =  T
 O3                               is not an invariant
 ERROR: tracer_cnst_init
```

### Will this change answers?

Yes

### Will you be implementing this yourself?

Yes, but I will need some help

### LINKED ISSUE #1191: Remove mct submodule
### Issue Type

Externals Update

### Issue Description

Remove the mct submodule, no longer used.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

### PR DESCRIPTION
**Pull request by jedwards4b**
This adds the C designation for concentration driven experiments, it's not clear to me if the emissions driven experiments will need the same date files or different ones so I've left that out for now.

Fixes #1077 
closes #1191 
closes #1190 
closes #1189

### CODE REVIEW

**Comment by jedwards4b** [comment]
@peverwhee I am going to need this change in alpha05b.  I need to also add changes for HIST as well as 1850 so I'll work on that and push it back here.  This is to support the C and E (Concentration and Emissions driven distinction) addition to compset names as documented in https://docs.google.com/spreadsheets/d/1nJl4lbbwBq60quyaERJ6OvVJxF4kDu4xX7zZenZgyB8/edit?usp=sharing.

**Comment by peverwhee** [comment]
ok thanks @jedwards4b. When you move this out of draft, I'll make it the next tag.

**Comment by jedwards4b** [comment]
I have added the C and E modifiers to compset names for Concentration and Emissions driven cases.  As we do not yet have Emissions driven compsets these are both currently pointing to the same cam use cases and stream files.

**Comment by peverwhee** [comment]
@jedwards4b is this tag waiting on a resolution to https://github.com/ESCOMP/CAM/issues/1189 ?

**Comment by jedwards4b** [comment]
I think that it would make sense for @fvitt 
to add the use case here.

**Comment by peverwhee** [comment]
@jedwards4b I'm going to kick off the tests, but can you confirm that Francis's use cases fix your error when you a get a chance?

**Review by peverwhee — APPROVED** [review]

**Comment by jedwards4b** [comment]
@peverwhee I have confirmed that test SMS_Ld2.ne30pg3_t232.B1850C_MTso.derecho_intel
now passes.

File: cime_config/config_component.xml [hunk 1]
```diff
@@ -392,9 +393,9 @@
     <type>char</type>
     <default_value>2000</default_value>
     <values match="last">
-      <value compset="^1850_">1850</value>
+      <value compset="^1850[CE]?_">1850</value>
```
Comment (fvitt) [inline cime_config/config_component.xml:396]: @peverwhee I think these changes to CAM_STREAM_NDEP settings are causing the differences in the namelist compare that you see in SMS_D_Ln9_P1280x1.ne30pg3_ne30pg3_mg17.FHISTC_MTt1s.derecho_intel.cam-outfrq9s_Leung_dust

File: bld/namelist_files/mozart_master_gas_drydep_list.xml [hunk 2]
```diff
@@ -77,7 +77,6 @@
  ISOPNOOHB
  ISOPNOOHD
  ISOPOH
- ISOPNO3
```
Comment (fvitt) [inline bld/namelist_files/mozart_master_gas_drydep_list.xml:80]: @peverwhee This change will cause base line failures in many chemistry compsets

File: cime_config/config_component.xml (hunk 1 above)
Comment (peverwhee) [reply cime_config/config_component.xml:396]: ah, ok. So would it be accurate to say it wasn't picking up the "right" settings for CAM_STREAM_NDEP_* until we added the `[CE]?` in this PR?

Comment (fvitt) [reply cime_config/config_component.xml:396]: Yes, I believe that is the case.

File: bld/namelist_files/mozart_master_gas_drydep_list.xml (hunk 2 above)
Comment (peverwhee) [reply bld/namelist_files/mozart_master_gas_drydep_list.xml:80]: that makes sense, just wanted to make sure the answer changes are expected/not TOO different

Comment (peverwhee) [reply bld/namelist_files/mozart_master_gas_drydep_list.xml:80]: @fvitt I feel like I'm not "nailing" communicating today! For clarity, are you OK with with answer changes caused by this PR?

Comment (fvitt) [reply bld/namelist_files/mozart_master_gas_drydep_list.xml:80]: Yes the answers changes are expected.  I am comfortable with answer changes in cases where the dep lists have changed in the namelist (removal of ISOPNO3).

Comment (peverwhee) [reply bld/namelist_files/mozart_master_gas_drydep_list.xml:80]: thank you francis!

**Pull request merged**